javascript/workflow/src/verbs/__tests__/binUtilities.spec.ts

```diff
@@ -124,4 +125,118 @@ describe('bin utilities tests', () => {
 			expect(result).toBe(35.70134669473044)
 		})
 	})
+
+	describe('huge numbers nice rounding test 1', () => {
+		const result = calculateNiceRounding(12000000, 78500000, 12389720.271622058)
+		const [min, max, width] = result
+
+		it('algorithm result', () => {
+			expect(min).toBe(10000000)
+			expect(max).toBe(80000000)
+			expect(width).toBe(10000000)
+		})
+	})
+
+	describe('huge numbers nice rounding test 2', () => {
+		const result = calculateNiceRounding(12000000, 78500000, 12671668.926380431)
+		const [min, max, width] = result
+
+		it('algorithm result', () => {
+			expect(min).toBe(10000000)
+			expect(max).toBe(80000000)
+			expect(width).toBe(10000000)
+		})
+	})
+
+	describe('huge numbers nice rounding test 3', () => {
+		const result = calculateNiceRounding(12000000, 78500000, 12460642.878078489)
+		const [min, max, width] = result
+
+		it('algorithm result', () => {
+			expect(min).toBe(10000000)
+			expect(max).toBe(80000000)
+			expect(width).toBe(10000000)
+		})
+	})
+
+	describe('regular numbers nice rounding test 1', () => {
+		const result = calculateNiceRounding(18, 67, 15.739759424941251)
```

Comment:
I think the minimum will always need to round down to make sure data values aren't excluded.